pkg/config/node.go

```diff
@@ -373,10 +416,11 @@ func getSortedBackends(kubeconfigPath string, readFromLocalAPI bool) (backends [
 		}).Info("Failed to get master Nodes list")
 		return []Backend{}, err
 	}
+	apiVipv6 := utils.IsIPv6(apiVip)
 	for _, node := range nodes.Items {
 		masterIp := ""
 		for _, address := range node.Status.Addresses {
-			if address.Type == v1.NodeInternalIP {
+			if address.Type == v1.NodeInternalIP && utils.IsIPv6(net.ParseIP(address.Address)) == apiVipv6 {
```

Comment:
you can also use [`k8s.io/utils.IsIPv6String()`](https://pkg.go.dev/k8s.io/utils/net#IsIPv6String) which takes a string as argument. So could also reduce some code in `util/addresses.go` But not sure if it's worth vendoring this package...